specs/extensions/openapi-extensions.md

```diff
@@ -127,4 +129,29 @@ paths:
                 - "I **should not** respond with either sample or full form of lyrics to the song because the song created by createSong(**params) is shown to the user directly along with the lyrics."
                 - "If createSong(**params) was not invoked because the service was overloaded or the user reached their invocation limit, I **must** tell the user they can visit the https://app.suno.ai website to create more songs and display the link using Markdown syntax."
                 - "I **shall** tell the user they can visit the https://app.suno.ai website to create more songs and display the link using Markdown syntax."
+```
+
+## x-openai-isConsequential
+Specifies whether a confirmation dialog should be displayed before running the operation. Possible values are:
+
+true: must always prompt the user for confirmation before running and don't show "Alaways allow" button
+false: it will show the prompt for confirmation with the "Always allow" button
+
+If the `x-openai-isConsequential` is not present, all GET operation will default to `false` and show the "Always allow" button; all other operations will default to `true` and must always prompt the user for confirmation.
```

Comment:
also, where does this go in the manifest?

Reply:
As a capability in the plugin function. https://spec-hub.azurewebsites.net/specifications/PluginManifest-2.1.html#example-6

Reply:
right, can we have the spec here updated with that information?

Reply:
This doesn't go to the manifest. This OAS is supported by Copilot, so we shouldn't touch it. When doing the slicing, we should just keep the information if present for the selected endpoint and that's it.
I'm not sure we should be creating confirmation object on behalf of users as I mentioned here: https://github.com/microsoft/kiota/pull/4961#discussion_r1674489038